admittance_controller/src/admittance_rule.cpp

```diff
@@ -240,22 +249,25 @@ controller_interface::return_type AdmittanceRule::update(
     }
   }
 
+  // TODO: Add here desired vector at the IK tip and not tool!!!!
```

Comment:
I'm going to add notes for several lines that aren't reachable from this PR.

Line 213 (above) seems unnecessary. Please double-check

Reply:
L227:

TODO: this formula could cause a "gimbal lock" if the angular difference is large

Reply:
Above L239, add comment:

// Acceleration is regarded as the control input. It is integrated to calculate velocity and acceleration.